File: java/hellouser/HelloUser.java
import java.util.Scanner;

public class HelloUser{

    public static void main(String [] args){
	Scanner scanner = new Scanner(System.in);

	System.out.print("Enter your name: ");
	String name = scanner.nextLine();

	System.out.println("Hello, "+name+"!");
    }
}
    

File: r/hellouser/hellouser.r
#This is for R interactive use
#name <- readline(prompt = "Enter your name: ")
# This is for command line use
# Usage: Rscript hellouser.r <name>
# Example: Rscript hellouser.r John
args <- commandArgs(trailingOnly = TRUE)
name <- args[1]
print(paste0("Hello, ", name, "!"))